Neos.Flow/Classes/Http/AbstractMessage.php

```diff
@@ -295,6 +297,185 @@ public function removeCookie($name)
+     */
+    public function getHeaderLine($name)
+    {
+        $headerLine = $this->headers->get($name);
+        if ($headerLine === null) {
+            $headerLine = '';
+        }
+
+        if (is_array($headerLine)) {
+            $headerLine = explode(', ', $headerLine);
+        }
+
+        return $headerLine;
+    }
+
+    /**
+     * Return an instance with the provided value replacing the specified header.
+     *
+     * While header names are case-insensitive, the casing of the header will
+     * be preserved by this function, and returned from getHeaders().
+     *
+     * This method MUST be implemented in such a way as to retain the
+     * immutability of the message, and MUST return an instance that has the
+     * new and/or updated header and value.
+     *
+     * @param string $name Case-insensitive header field name.
+     * @param string|string[] $value Header value(s).
+     * @return self
+     * @throws \InvalidArgumentException for invalid header names or values.
+     */
+    public function withHeader($name, $value)
+    {
+        $newMessage = clone $this;
+        $newMessage->setHeader($name, $value, true);
+        return $newMessage;
+    }
+
+    /**
+     * Return an instance with the specified header appended with the given value.
+     *
+     * Existing values for the specified header will be maintained. The new
+     * value(s) will be appended to the existing list. If the header did not
+     * exist previously, it will be added.
+     *
+     * This method MUST be implemented in such a way as to retain the
+     * immutability of the message, and MUST return an instance that has the
+     * new header and/or value.
+     *
+     * @param string $name Case-insensitive header field name to add.
+     * @param string|string[] $value Header value(s).
+     * @return self
+     * @throws \InvalidArgumentException for invalid header names or values.
+     */
+    public function withAddedHeader($name, $value)
+    {
+        $newMessage = clone $this;
+        $newMessage->setHeader($name, $value);
+
+        return $newMessage;
+    }
+
+    /**
+     * Return an instance without the specified header.
+     *
+     * Header resolution MUST be done without case-sensitivity.
+     *
+     * This method MUST be implemented in such a way as to retain the
+     * immutability of the message, and MUST return an instance that removes
+     * the named header.
+     *
+     * @param string $name Case-insensitive header field name to remove.
+     * @return self
+     */
+    public function withoutHeader($name)
+    {
+        $newMessage = clone $this;
+        $newMessage->getHeaders()->remove($name);
+
+        return $newMessage;
+    }
+
+    /**
+     * Gets the body of the message.
+     *
+     * @return StreamInterface Returns the body as a stream.
+     */
+    public function getBody()
+    {
+        $streamResource = fopen('php://memory', 'r+');
```

Comment:
Sure, but please let's make iterations. As long as there is nothing broken I would first like to get this in...